linenoise.c

```diff
@@ -1168,7 +1170,11 @@ int linenoiseHistorySave(const char *filename) {
     fp = fopen(filename,"w");
     umask(old_umask);
     if (fp == NULL) return -1;
-    chmod(filename,S_IRUSR|S_IWUSR);
+    if (chmod(filename,S_IRUSR|S_IWUSR) < 0) {
```

Comment:
> So what happens in the very odd case where we can get a file handle but can't immediately run a system call on it?

Theoretically, `fopen()` and `chmod()` can be replaced with a combination of `open()`, `chmod()` and `fdopen()`, but I can't think of a good reason to do so.